client/app/components/EventButtons.tsx

```diff
@@ -13,19 +13,19 @@ const EventButtons = ({
 }: {
   eventId: string;
   events: IEvent[];
-  forPage: 'results' | 'rankings' | 'data-entry';
+  forPage: 'results' | 'rankings' | 'competitions' | 'data-entry';
 }) => {
   const router = useRouter();
   const { id, singleOrAvg } = useParams();
   const searchParams = useSearchParams();
 
   const [selectedCat, setSelectedCat] = useState(
-    eventCategories.find((el) => events.find((e) => e.eventId === eventId).groups.includes(el.group)),
+    eventCategories.find((el) => events.find((e) => e.eventId === eventId)?.groups.includes(el.group)),
   );
 
   // If hideCategories = true, just show all events that were passed in
   const filteredEvents = useMemo(
-    () => (forPage !== 'rankings' ? events : events.filter((el) => el.groups.includes(selectedCat.group))),
+    () => (['rankings', 'competitions'].includes(forPage) ? events : events.filter((el) => el.groups.includes(selectedCat.group))),
```

Comment:
There should be a ! before the [ here, or you can swap the expressions after ? and :